modules/repository/generate.go

```diff
@@ -353,3 +362,12 @@ func GenerateRepository(ctx context.Context, doer, owner *user_model.User, templ
 
 	return generateRepo, nil
 }
+
+// Escapes user input to valid OS filenames
+//
+//	https://github.com/sindresorhus/filename-reserved-regex
+func fileNameEscape(s string) string {
+	re := regexp.MustCompile(`(?i)[<>:\"/\\|?*\x{0000}-\x{001F}]|^(con|prn|aux|nul|com\d|lpt\d)$`)
```

Comment:
Can we protect against and test the case where the substitution results in `.` or `..`?  I don't know what will happen if we try to rename something to that, but it seems like it might be safer to rule that out here, rather than risk some kind of unexpected directory traversal security issue.

Reply:
We can if you think it’s a good idea or I can at test it. We already substitute any forward and backslashes, but maybe if you can get a path to substitute at .. it could cause an issue. Probably a good idea.